examples/gke-swap/python-density/run_pythonsandbox_density_test.sh

```
@@ -0,0 +1,187 @@
+    for density in ${DENSITIES}; do
+        log "----------------------------------------------------------------------"
+        log "Starting sweep: Pool=${pool}, Density=${density}"
+        log "----------------------------------------------------------------------"
+
+        # Create sweep-specific output directory for metrics and test logs
+        SWEEP_DIR="${ARTIFACT_DIR}/${pool}/${density}"
+        mkdir -p "${SWEEP_DIR}"
+
+        # Clean up any lingering test namespaces, sandboxes, or finalizers from prior failed runs
+        log "Purging any lingering test namespaces and stripping finalizers..."
+        kubectl get sandboxes -A -o jsonpath='{range .items[*]}{.metadata.namespace}{" "}{.metadata.name}{"\n"}{end}' 2>/dev/null | xargs -L1 -r bash -c 'kubectl patch sandbox $1 -n $0 -p "{\"metadata\":{\"finalizers\":[]}}" --type=merge 2>/dev/null' || true
```

Shellcheck (0.11.0)

[info] Expressions don't expand in single quotes, use double quotes for that.

(SC2016)